src/Model/Behavior/SocialBehavior.php

```diff
@@ -276,19 +277,17 @@ public function generateUniqueUsername($username)
      * Prepare a query to retrieve existing entity for social login
      *
      * @param \Cake\ORM\Query\SelectQuery $query The base query.
-     * @param array $options Find options with email key.
+     * @param mixed $email Find options with email key.
      * @return \Cake\ORM\Query\SelectQuery
      */
-    public function findExistingForSocialLogin(\Cake\ORM\Query\SelectQuery $query, array $options)
+    public function findExistingForSocialLogin(SelectQuery $query, mixed $email): SelectQuery
     {
-        $email = $options['email'] ?? null;
-        if (!$email) {
-            return $query->where('1 != 1');
-        }
-
-        return $query->where([
-            $this->_table->aliasField('email') => $email,
-        ]);
+        return $query->where(
+            fn(QueryExpression $expression) => $expression->eq(
```

Comment:
why the complicated query expression?

Reply:
Thanks for pointing this out, i restored the old, simple way